src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -614,142 +615,197 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    .rs.fromJSON(.rs.base64decode(encoded))
 })
 
-.rs.addFunction("parseYamlOpt", function(opt)
+
+.rs.addFunction("parseYamlOptImpl", function(opt)
 {
-    opt <- sub("^#\\|\\s*", "", opt)
-    if (any(grepl(".+=.+", opt))) # R-style chunk options
-    {
-        opts <- paste(opt, collapse = " ")
-        opts <- knitr:::parse_params(opts)
-    }
-    else
-    {
-        opts <- .rs.fromYAML(opt)
-        # parse truthy and falsy style yaml options
-        opts <- lapply(opts, function(value)
-           {
-             if (!is.character(value))
-               return(value)
-             if (tolower(value) %in% c("y", "yes", "on"))
-               return(TRUE)
-             if (tolower(value) %in% c("n", "no", "off"))
-               return(FALSE)
-             else
-               return(value)
-           })
-    }
-    return(opts)
+   # read as YAML
+   opts <- .rs.fromYAML(opt)
+   
+   # parse truthy and falsy style yaml options
+   opts <- lapply(opts, function(value)
+   {
+      if (!is.character(value))
+         return(value)
+      if (tolower(value) %in% c("y", "yes", "on"))
+         return(TRUE)
+      if (tolower(value) %in% c("n", "no", "off"))
+         return(FALSE)
+      else
+         return(value)
+   })
+   
+   opts
 })
 
-.rs.addFunction("evaluateChunkOptions", function(code)
+.rs.addFunction("parseYamlOpt", function(opt, commentPrefix = "#")
```

Comment:
is the comment character ever not `#`?

Reply:
Okay, I see below we have alternatives for some engines like `"cc", "cpp", "go", "js", "node", "rcpp", "stan"` etc (which I haven't used, only seen the r python julia examples) but wondering why down below we set this as `#+` but here the default arg we use `#`.  Are multiple `#` allowed in this context?

Reply:
Good question... it looks like only a single instance of the comment character is allowed, so e.g.

```
#| label: foo     # okay
##| label: foo    # not okay
```

I think I need to tweak the other usages then.